server/bin/pbench-backup-tarballs.py

```diff
@@ -0,0 +1,485 @@
+        self.status = status
+        self.ntotal = ntotal
+        self.nbackup_success = nbackup_success
+        self.nbackup_fail = nbackup_fail
+        self.ns3_success = ns3_success
+        self.ns3_fail = ns3_fail
+        self.nquaran = nquaran
+
+
+def sanity_check(s3_obj, config, logger):
+
+    # make sure archive is present
+    archive = config.ARCHIVE
+
+    if not os.path.realpath(archive):
+        logger.error(
+            'The ARCHIVE directory {}, does not resolve to a real location'.format(archive))
+        return 1
+
+    if not os.path.isdir(archive):
+        logger.error(
+            'The ARCHIVE directory {}, does not resolve {} to a directory'.format(archive, os.path.realpath(archive)))
+        return 1
+
+    # make sure the local backup directory is present
+    backup = config.BACKUP
+
+    if len(backup) == 0:
+        logger.error(
+            'Unspecified backup directory, no pbench-backup-dir config in pbench-server section')
+        return 1
+
+    try:
+        os.mkdir(backup)
+    except FileExistsError:
+        # directory already exists, verify it
+        if not os.path.realpath(backup):
+            logger.error(
+                'The BACKUP directory {}, does not resolve to a real location'.format(backup))
+            return 1
+
+        if not os.path.isdir(backup):
+            logger.error(
+                'The BACKUP directory {}, does not resolve {} to a directory'.format(backup, os.path.realpath(backup)))
+            return 1
+    except Exception:
+        logger.exception(
+            "os.mkdir: Unable to create backup destination directory: {}\n".format(backup))
+        return 1
+
+    # make sure the quarantine directory is present
+    qdir = config.QDIR
+
+    if len(qdir) == 0:
+        _logger.error(
+            'Unspecified quarantine directory, no pbench-quarantine-dir config in pbench-server section')
+        return 1
+
+    if not os.path.realpath(qdir):
+        _logger.error(
+            'The QUARANTINE directory {}, does not resolve to a real location'.format(qdir))
+        return 1
+
+    if not os.path.isdir(qdir):
+        _logger.error(
+            'The QUARANTINE directory {}, does not resolve {} to a directory'.format(qdir, os.path.realpath(qdir)))
+        return 1
+
+    # make sure the S3 bucket exists
+    try:
+        s3_obj.connector.head_bucket(Bucket='{}'.format(s3_obj.bucket_name))
+    except Exception as e:
+        _logger.exception(
+            "Bucket: {} does not exist or you have no access. {}\n".format(s3_obj.bucket_name, e))
+        return 1
+
+
+def prepare_report(report, TS, prog, config, result_string):
+    with open(report, 'a') as f:
+        f.write("{}.{}({})\n{}\n".format(
+            prog, TS, config.PBENCH_ENV, result_string))
+
+
+def backup_to_local(config, logger, controller_path, controller, tb, tar, resultname, archive_md5, archive_md5_hex_value):
+
+    nbackup_success = nbackup_fail = 0
```

Comment:
These variables are not needed: the two functions dealing with a single tarball can only have (0, 1) values for these.